### PR TITLE
[vcpkg.cmake] Fix toolchain

### PR DESCRIPTION
**Pull request by Neumann-A**
includes #23410

cc @JackBoosY

### CODE REVIEW

**Comment by JackBoosY** [comment]
Testing...
Can we merge this PR's changes to https://github.com/microsoft/vcpkg/pull/23410?

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -518,17 +518,22 @@ if(VCPKG_MANIFEST_MODE AND VCPKG_MANIFEST_INSTALL AND NOT Z_VCPKG_CMAKE_IN_TRY_C
...
+unset(VCPKG_CAN_USE_HOST_TOOLS)
 
 if(VCPKG_SETUP_CMAKE_PROGRAM_PATH)
     set(tools_base_path "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/tools")
     if(VCPKG_USE_HOST_TOOLS)
         set(tools_base_path "${VCPKG_INSTALLED_DIR}/${VCPKG_HOST_TRIPLET}/tools")
     endif()
     list(APPEND CMAKE_PROGRAM_PATH "${tools_base_path}")
-    file(GLOB_RECURSE Z_VCPKG_TOOLS_DIRS "${tools_base_path}/*")
+    file(GLOB_RECURSE Z_VCPKG_TOOLS_DIRS LIST_DIRECTORIES true "${tools_base_path}/*")
```
[the remark this answers is not in the document]
Comment (JackBoosY) [reply scripts/buildsystems/vcpkg.cmake:534]: Using `GLOB_RECURSE` will significantly increase the configuration time. According to the policy, all executables should be installed in _tools/${PORT}_, so do we need to recursively look for these executable directories here?

**Comment by Neumann-A** [comment]
> Can we merge this PR's changes to https://github.com/microsoft/vcpkg/pull/23410?

You can do that if you want. I won't mind. 

I am still suprised that toolchain changes get merged without a full CI rebuild....

**Comment by JackBoosY** [comment]
This changes can fix the issue.

File: scripts/buildsystems/vcpkg.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (Neumann-A) [reply scripts/buildsystems/vcpkg.cmake:534]: >  so do we need to recursively look for these executable directories here?

Yes we do..... look at the file list. Qt stuff for examples gets installed into tools/Qt6/bin etc. 

Also globbing tools is not that slow.... compiler invokes/configure checks are much slower....

**Comment by JackBoosY** [comment]
> > Can we merge this PR's changes to #23410?
> 
> You can do that if you want. I won't mind.
> 
> I am still suprised that toolchain changes get merged without a full CI rebuild....

Perhaps this was due to our pipeline being broken and retriggering the test did not effectively cover this modification.

**Comment by Neumann-A** [comment]
> Perhaps this was due to our pipeline being broken and retriggering the test did not effectively cover this modification.

Na it wasn't tested because it would have come up. Not all pipelines are broken.

**Comment by JackBoosY** [comment]
Okay, merged this changes into https://github.com/microsoft/vcpkg/pull/23410.

Thanks!

**Pull request closed without merging**